module/zfs/vdev_queue.c

```diff
@@ -246,6 +246,14 @@ vdev_queue_offset_compare(const void *x1, const void *x2)
 	if (likely(cmp))
 		return (cmp);
 
+	/*
+	 * If the zio we are trying to find is requesting
+	 * an exact offset match then return once TREE_CMP finds
+	 * the offset.
+	 */
+	if (z1->io_offset_match)
```

Comment:
Are you sure that z1 (not z2) is the one we're trying to find?  Is that part of the libavl API?  I think this would be more clear/foolproof: 
```suggestion
	if (z1->io_offset_match || z2->io_offset_match)
```